### PR TITLE
Move dart-services to serving Dart 2.0 by default

### PR DESCRIPTION
**Pull request by jcollins-g**
This doesn't strip out all the Dart 1 support just yet, but it should work for Dart 2.0.

### CODE REVIEW

**Comment by jcollins-g** [comment]
I'm going to set up a dart-pad front-end and backend test version so we can demo and exercise this before landing.

**Comment by jcollins-g** [comment]
https://dart2-test-dot-dart-pad.appspot.com

**Comment by devoncarew** [comment]
Optional new works :)

https://dart2-test-dot-dart-pad.appspot.com/24b488717dfee808bcd8f535db191f08

File: lib/src/analysis_server.dart [hunk 1]
```diff
@@ -40,16 +41,19 @@ class AnalysisServerWrapper {
   /// Instance to handle communication with the server.
   AnalysisServer analysisServer;
 
-  AnalysisServerWrapper(this.sdkPath, {bool strongMode: true}) {
-    _logger.info('AnalysisServerWrapper ctor, strong mode: $strongMode');
+  AnalysisServerWrapper(this.sdkPath,
+      {bool strongMode: true, this.previewDart2: false}) {
```
Comment (devoncarew) [inline lib/src/analysis_server.dart]: Not for this PR, but we can likely remove all `strongMode` flags, and won't need to generate an analysis_options file below, just to enable strong mode.

File: lib/src/common_server.dart [hunk 2]
```diff
@@ -75,12 +74,13 @@ class CommonServer {
     pub = enablePackages ? new Pub() : new Pub.mock();
     compiler = new Compiler(sdkPath, pub);
     analysisServer = new AnalysisServerWrapper(sdkPath, strongMode: false);
-    analysisServerStrong = new AnalysisServerWrapper(sdkPath);
+    analysisServerStrong =
```
Comment (devoncarew) [inline lib/src/common_server.dart]: After this PR, we can move to only have one analysis server instance (we had two to support both strong and on-strong).

File: lib/src/compiler.dart [hunk 3]
```diff
@@ -66,7 +68,7 @@ class Compiler {
         '--suppress-hints',
         '--terse',
       ];
-
+      if (previewDart2) arguments.add('--preview-dart-2');
       if (useCheckedMode) arguments.add('--checked');
```
Comment (devoncarew) [inline lib/src/compiler.dart]: @sigmundch, does dart2js support `--enable-asserts` (like the VM fro Dart 2), or still use `--checked`?

Comment (devoncarew) [reply lib/src/compiler.dart]: And I wonder if generally we should compile dartpad snippets with enable asserts / checked mode?

Comment (sigmundch) [reply lib/src/compiler.dart]: Yes, we also support `--enable-asserts` - after we delete our support for Dart1, we expect to get rid of `--checked` entirely, so I do recommend switching to it.

Re default: I guess it makes sense to have it on by default in Dartpad - I expect that if users write an assert in their dartpad snippet, they most likely want it to get executed.

Comment (jcollins-g) [reply lib/src/compiler.dart]: The checked mode flag comes from the request; dart-services could start ignoring it and just force it to be true all the time.  I'll convert the checked mode to use --enable-asserts to get dart-services ahead of the dart2js's plans (thanks @sigmundch !)

**Comment by jcollins-g** [comment]
PTAL: now supports --enable-asserts.  Also I changed the `Logger` instance to be a public variable so that I could switch on logging to stdout for the common_server unit tests (making debugging easier).

**Pull request merged**